Flaky tests possibly related to authentication or rate limiting

**Describe the bug**

As the title says, some tests are flaky and the errors are sufficiently swallowed and not logged so that there is no indication what is actually going wrong.

We should probably refactor the tests to not use requests to generate users but to instead use fixtures. This will speed the tests up significantly too as a nice bonus.

Example:
https://github.com/scidsg/hushline/actions/runs/10942529787/job/30379765896?pr=588

```
=================================== FAILURES ===================================
_____________________________ test_change_password _____________________________

client = <FlaskClient <Flask 'hushline'>>

    def test_change_password(client: FlaskClient) -> None:
        username = "test_change_password"
        original_password = f"{token_urlsafe(16)}!"
        new_password = f"{token_urlsafe(16)}!!!"
    
>       user = register_user(client, username, original_password)

tests/test_settings.py:75: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

client = <FlaskClient <Flask 'hushline'>>, username = 'test_change_password'
password = 'FznbXfIrnRAUOblBGnKuUw!'

    def register_user(client: FlaskClient, username: str, password: str) -> User:
        # Prepare the environment to not require invite codes
        os.environ["REGISTRATION_CODES_REQUIRED"] = "False"
    
        user_data = {"username": username, "password": password}
    
        response = client.post("/register", data=user_data, follow_redirects=True)
        assert response.status_code == 200
>       assert b"Registration successful!" in response.data
E       AssertionError

tests/auth_helper.py:20: AssertionError
```

--- a/hushline/routes.py
+++ b/hushline/routes.py
@@ -475,7 +475,7 @@ def init_app(app: Flask) -> None:
                 flash("⏲️ Please wait a moment before trying again.")
                 return render_template("verify_2fa_login.html", form=form), 429
 
-            if totp.verify(verification_code):
+            if totp.verify(verification_code, valid_window=1):
                 auth_log = AuthenticationLog(
                     user_id=user.id, successful=True, otp_code=verification_code, timecode=timecode
                 )

--- a/hushline/settings/__init__.py
+++ b/hushline/settings/__init__.py
@@ -400,7 +400,7 @@ def create_blueprint() -> Blueprint:
             if (
                 verification_code
                 and temp_totp_secret
-                and pyotp.TOTP(temp_totp_secret).verify(verification_code)
+                and pyotp.TOTP(temp_totp_secret).verify(verification_code, valid_window=1)
                 and user
             ):
                 user.totp_secret = temp_totp_secret
@@ -465,7 +465,7 @@ def create_blueprint() -> Blueprint:
 
         verification_code = request.form["verification_code"]
         totp = pyotp.TOTP(user.totp_secret)
-        if not totp.verify(verification_code):
+        if not totp.verify(verification_code, valid_window=1):
             flash("⛔️ Invalid 2FA code. Please try again.")
             return redirect(url_for("show_qr_code"))
 

